resources/build/zip.inc.sh

```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env bash
+#
+# This script contains zip and unzip (TODO) utilities to function for all environments.
+# unzip is available in all environments
+# zip is not available by default on Windows, but can be manually installed.
+# If zip is not available, then fall back to 7z for zipping.
+# 7z requires env SEVENZ_HOME.
+#
+# TODO: refactor with /resources/build/win/zip.inc.sh
+
+# zip/7z to create an archive with the following parameters (in order)
+# [zip filename]
+# [filename of files to exclude in the zip]
+# [list of flags to pass to zip command, 'a' or leading with single-dash]
+# [list of files to include in zip]
+function zip_files() {
+
+  # Parse parameters
+
+  # $1 for zip filename
+  local ZIP_FILE="$1"
+
+  # $2 for filename containing files to exclude
+  local EXCLUDE="$2"
+  shift 2
+  if ! test -f ${EXCLUDE}; then
+    builder_die "Filename of files to exclude '${EXCLUDE}' does not exist"
+  fi
+
+  # Parse rest of parameters
+  FLAGS=()
+  INCLUDE=()
+  while [[ $# -gt 0 ]] ; do
+    case "$1" in
+      a|-*)
+        FLAGS+=($1)
+        shift
+        ;;
+
+      *)
+        INCLUDE+=($1)
+        shift
+        ;;
+    esac
+  done
```

Comment:
`a` is implicit for 7z and illegal for zip -- so this doesn't work.

I would prefer to use the 'standard' way of parsing flags:

1. while parameter starts with `-`, treat it as a flag.
2. first parameter to not start with `-` is end of flags and start of files
3. `--` is a separator parameter that also ends flags.

Reply:
If I do `-x@filename`, do we no longer need `--`?

e.g. When `-x@filename` not given, there's no files to exclude.

Reply:
We probably don't need to separate files and flags -- in the usual case?